src/AddressId.sol

```diff
@@ -118,8 +121,8 @@ contract AddressId {
 
     function _transferFromCaller(IERC20 erc20, uint128 amt) internal {
         require(erc20.transferFrom(msg.sender, address(this), amt), "Transfer from caller failed");
-        if (erc20.allowance(address(this), address(dripsHub)) < amt) {
-            erc20.approve(address(dripsHub), type(uint256).max);
+        if (erc20.allowance(address(this), reserve) < amt) {
+            erc20.approve(reserve, type(uint256).max);
```

Comment:
This pattern together with removing the `msg.sender` from the reserve is a bit of a security issue.

It would be somehow dangerous to use the dripsHub as an EOA.

Reply:
Could you elaborate about the security impact? I don't understand the `use the dripsHub as an EOA` part.

Reply:
If user want to generate subAccounts with an EOA address for whatever reason.
In that case If I understood correctly an approval on the reserve would be required.

However, I agree we have the same problem with Governance upgradability.